Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerGroupStore.cs

```diff
@@ -341,6 +356,65 @@ public async Task<Group> DeleteUserFromGroup(Group group, User user)
             return group;
         }
 
+        public async Task<Group> AddChildGroups(Group group, IEnumerable<Group> childGroups)
+        {
+            foreach (var childGroup in childGroups)
+            {
+                var newChildGroup = new ChildGroup
+                {
+                    ParentGroupId = group.Id,
+                    ChildGroupId = childGroup.Id
+                };
+
+                AuthorizationDbContext.ChildGroups.Add(newChildGroup);
+                group.Children.Add(childGroup);
+            }
+
+            await AuthorizationDbContext.SaveChangesAsync();
+            await EventService.RaiseEventAsync(new EntityAuditEvent<Group>(EventTypes.ChildEntityCreatedEvent, group.Id.ToString(), group));
+
+            return group;
+        }
+
+        public async Task<Group> RemoveChildGroups(Group group, IEnumerable<Group> childGroups)
+        {
+            var childGroupEntities = AuthorizationDbContext.ChildGroups.Where(cg => !cg.IsDeleted && cg.ParentGroupId == group.Id && childGroups.Select(g => g.Id).Contains(cg.ChildGroupId));
+            foreach (var childGroup in childGroups)
+            {
+                var childToRemove = group.Children.FirstOrDefault(c => c.Name == childGroup.Name);
+                if (childToRemove == null)
+                {
+                    continue;
+                }
+
+                group.Children.Remove(childToRemove);
+
+                var childToRemoveEntity = childGroupEntities.FirstOrDefault(cg => cg.ParentGroupId == childGroup.Id && cg.ChildGroupId == group.Id);
+                childToRemoveEntity.IsDeleted = true;
+            }
+
+            await AuthorizationDbContext.SaveChangesAsync();
+            await EventService.RaiseEventAsync(new EntityAuditEvent<Group>(EventTypes.ChildEntityDeletedEvent, group.Id.ToString(), group));
+
+            return group;
+        }
+
+        public Task<IEnumerable<Group>> Get(IEnumerable<string> childGroupNames)
```

Comment:
Should we name this parameter something other than childGroupNames? Seems like this function is good at getting a list of groups based on a list of group names and is not really specific to childGroups...

Reply:
Good call. Will change it.